cluster-autoscaler/core/scale_down.go

```diff
@@ -266,6 +268,11 @@ func (sd *ScaleDown) TryToScaleDown(nodes []*apiv1.Node, pods []*apiv1.Pod, pdbs
 	// to recreate on other nodes.
 	emptyNodes := getEmptyNodes(candidates, pods, sd.context.MaxEmptyBulkDelete, sd.context.CloudProvider)
 	if len(emptyNodes) > 0 {
+		nodeDeletionStart := time.Now()
+		defer func(nodeDeletionStart time.Time, nodeDeletionDuration *time.Duration) {
+			stop := time.Now()
+			*nodeDeletionDuration = stop.Sub(nodeDeletionStart)
+		}(nodeDeletionStart, &nodeDeletionDuration)
```

Comment:
Only reason was it felt a little scary to spread the duration start and stop in the code so much as it will be very easy to put a little return between them and mess with the setup. But I guess hacking around this might not be the best way.